src/main/java/com/google/neighborgood/servlets/MessageServlet.java

```diff
@@ -105,6 +107,42 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     msgEntity.setProperty("sentTime", System.currentTimeMillis());
 
     datastore.put(msgEntity);
+
+    // After storing the message, create a notification entity to notify the receiver
+    Key taskKey = KeyFactory.stringToKey(taskId);
+
+    Entity taskEntity;
+    try {
+      taskEntity = datastore.get(taskKey);
+    } catch (EntityNotFoundException e) {
+      System.err.println("Unable to find the entity based on the input key");
+      response.sendError(HttpServletResponse.SC_NOT_FOUND, "The requested task could not be found");
+      return;
+    }
+
+    String owner = (String) taskEntity.getProperty("Owner");
+    String helper = (String) taskEntity.getProperty("Helper");
+    String currentUser = userService.getCurrentUser().getUserId();
+
+    Entity notificationEntity;
+    if (currentUser.equals(helper)) {
+      notificationEntity = new Entity("Notification");
+      notificationEntity.setProperty("receiver", owner);
+      notificationEntity.setProperty("taskId", taskId);
+      datastore.put(notificationEntity);
+    } else if (currentUser.equals(owner)) {
+      // If the task is still OPEN, we would not send a notification
```

Comment:
Edge case: what if the task is open, the owner writes messages, and then the task is picked up by a helper, should the helper see a notification at that point?

Reply:
Originally I didn't design the feature since it's a little bit hard to achieve. But I think it's a really good idea so I add this feature in the new commit.